packages/grpc-js/src/server-call.ts

```diff
@@ -146,3 +253,218 @@ export type Handler<RequestType, ResponseType> = {
+    const headers = Object.assign(defaultResponseHeaders, custom);
+    this.stream.respond(headers, defaultResponseOptions);
+  }
+
+  receiveMetadata(headers: http2.IncomingHttpHeaders) {
+    const metadata = Metadata.fromHttp2Headers(headers);
+
+    // TODO(cjihrig): Receive compression metadata.
+
+    const timeoutHeader = metadata.get(GRPC_TIMEOUT_HEADER);
+
+    if (timeoutHeader.length > 0) {
+      const match = timeoutHeader[0].toString().match(DEADLINE_REGEX);
+
+      if (match === null) {
+        const err = new Error('Invalid deadline') as ServiceError;
+        err.code = Status.OUT_OF_RANGE;
+        this.sendError(err);
+        return;
+      }
+
+      const timeout = (+match[1] * deadlineUnitsToMs[match[2]]) | 0;
+
+      this.deadline = setTimeout(handleExpiredDeadline, timeout, this);
+      metadata.remove(GRPC_TIMEOUT_HEADER);
+    }
+
+    return metadata;
+  }
+
+  receiveUnaryMessage(): Promise<RequestType> {
+    return new Promise((resolve, reject) => {
+      const stream = this.stream;
+      const chunks: Buffer[] = [];
+      let totalLength = 0;
+
+      stream.on('data', (data: Buffer) => {
+        chunks.push(data);
+        totalLength += data.byteLength;
+      });
+
+      stream.once('end', async () => {
+        try {
+          const requestBytes = Buffer.concat(chunks, totalLength);
+
+          resolve(await this.deserializeMessage(requestBytes));
+        } catch (err) {
+          err.code = Status.INTERNAL;
+          this.sendError(err);
+          resolve();
+        }
+      });
+    });
+  }
+
+  serializeMessage(value: ResponseType) {
+    const handler = this.handler as Handler<RequestType, ResponseType>;
+    const messageBuffer = handler.serialize(value);
+
+    // TODO(cjihrig): Call compression aware serializeMessage().
+    const byteLength = messageBuffer.byteLength;
+    const output = Buffer.allocUnsafe(byteLength + 5);
+    output.writeUInt8(0, 0);
+    output.writeUInt32BE(byteLength, 1);
+    messageBuffer.copy(output, 5);
+    return output;
+  }
+
+  async deserializeMessage(bytes: Buffer) {
+    const handler = this.handler as Handler<RequestType, ResponseType>;
+    // TODO(cjihrig): Call compression aware deserializeMessage().
+    const receivedMessage = bytes.slice(5);
+
+    return handler.deserialize(receivedMessage);
+  }
+
+  async sendUnaryMessage(
+      err: ServiceError|null, value: ResponseType|null, metadata?: Metadata,
+      flags?: number) {
+    if (err) {
+      if (metadata) {
+        err.metadata = metadata;
+      }
+
+      this.sendError(err);
+      return;
+    }
+
+    try {
+      const response = await this.serializeMessage(value as ResponseType);
+
+      this.write(response);
+      this.sendStatus(
+          {code: Status.OK, details: 'OK', metadata} as StatusObject);
+    } catch (err) {
+      err.code = Status.INTERNAL;
+      this.sendError(err);
+    }
+  }
+
+  sendStatus(statusObj: StatusObject) {
+    if (this.cancelled === true) {
```

Comment:
`this.cancelled` is a boolean. You don't need to explicitly compare it with `true`.